32-explore_pathway_separation.Rmd

```diff
@@ -0,0 +1,416 @@
+```
+
+We're going to represent this as a heatmap, so we'll need to wrangle the
+results into a `data.frame` that looks like this
+
+|   | 500 | ... | 32000 |
+|:-:|:---:|:---:|:-----:|
+|IFN|  0  | ... | 3     |
+|MYELOID|  0  | ... | 5   |
+
+Where we're counting how many of the 5 models (repeats) the pairs of pathways
+are separated.
+
+```{r}
+size.df <- size.results.df %>%
+  # for each sample_size, pathway pair
+  dplyr::group_by(sample_size, pathway) %>%
+  # count how many models where there's separation
+  dplyr::summarize(model_count = sum(value)) %>%
+  # spread the columns
+  tidyr::spread(sample_size, model_count) %>%
+  # reorder such that sample size is increasing
+  dplyr::select(c("pathway", "500", "1000", "2000", "4000", "8000", 
+                  "16000", "32000")) %>%
+  # the pathway names should be rownames rather than an individual column
+  tibble::column_to_rownames("pathway") %>%
+  as.data.frame()
+
+# let's take a look at the resulting data.frame
+size.df
+```
+
+```{r}
+size.heatmap <- 
+  ComplexHeatmap::Heatmap(as.matrix(size.df),
+                          cluster_rows = FALSE,
+                          cluster_columns = FALSE,
+                          row_names_side = "left",
+                          column_names_side = "top",
+                          col = colorRampPalette(c("white", "blue3"))(6),
+                          rect_gp = grid::gpar(col = "black"),
+                          show_heatmap_legend = TRUE,
+                          name = "number of models")
+size.heatmap
+```
+
+## Biological context
+
+```{r}
+# detect for pathway separation
+context.results <- lapply(context.model.files,  # for each context (5x repeat)
+                       function(x) {
+                         # read in the list of 5 models
+                         models.list <- readRDS(x)
+                         lapply(models.list, 
+                                function(y) {
+                                  # we need to specifically extract the `PLIER`
+                                  # element of the list and test all pairs
+                                  # of pathways
+                                  AllPairs(plier.model = y$PLIER)
+                                })
+                       })
+
+# we'll name the elements of the list based on their training set, which we 
+# can extract from the filenames themselves
+names(context.results) <- 
+  stringr::str_match(context.model.files, "recount2_(.*?)_accessions")[, 2]
+```
+
+```{r}
+# melt and bind the pathway separation results, using the sample size as an
+# identifier
+context.results.df <- dplyr::bind_rows(lapply(context.results, 
+                                           reshape2::melt), 
+                                    .id = "biological_context")
+colnames(context.results.df)[3:4] <- c("pathway", "seed")
+```
+
+Wrangle data for heatmap
+
+```{r}
+context.df <- context.results.df %>%
```

Comment:
can think about making this wrangling into a function since it is repeated above. dplyr [plays nicely](https://sebastiansauer.github.io/prop_fav/) in functions now

Reply:
Thanks! I needed to use `rlang::sym` instead of `rlang::enquo` (https://github.com/greenelab/multi-plier/pull/52/commits/e9b5682ab98f3e9a5e3c8f5638ab88b75d2c11cb) but this got me started in the right direction.